Userland/Utilities/cat.cpp

```diff
@@ -23,42 +22,23 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.add_positional_argument(paths, "File path", "path", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    Vector<int> fds;
-    if (paths.is_empty()) {
-        TRY(fds.try_append(STDIN_FILENO));
-    } else {
-        for (auto const& path : paths) {
-            int fd;
-            if (path == "-") {
-                fd = 0;
-            } else {
-                auto fd_or_error = Core::System::open(path, O_RDONLY);
-                if (fd_or_error.is_error()) {
-                    warnln("Failed to open {}: {}", path, fd_or_error.error());
-                    continue;
-                }
-                fd = fd_or_error.release_value();
-            }
-            TRY(fds.try_append(fd));
-        }
-    }
+    if (paths.is_empty())
+        paths.append("-"sv);
+
+    Vector<NonnullOwnPtr<Core::Stream::File>> files;
+    TRY(files.try_ensure_capacity(paths.size()));
+
+    for (auto const& path : paths)
+        files.unchecked_append(TRY(Core::Stream::File::open_file_or_standard_stream(path, Core::Stream::OpenMode::Read)));
```

Comment:
Ah sorry, I just realized there is a bit of change of behavior here. If any file can't be opened, we abort now. Previously, we would report an error and move on to valid files.

Linux behaves the same way - if I specify 2 files and 1 doesn't exist, it still prints the existing file, and just a warning for the bad file. So maybe this should continue to manually check for an error and just report it:
```c++
for (auto const& path : paths) {
    auto file_or_error = Core::Stream::File::open_file_or_standard_stream(path, Core::Stream::OpenMode::Read);
    if (file_or_error.is_error())
        warnln("Failed to open {}: {}", path, file_or_error.error());
    else
        files.unchecked_append(file_or_error.release_value());
}
```

There's also an existing bug in the old implementation where if we couldn't open a file, we still return 0 at the end. So if you want, a free commit might be to count the number of files that can't be opened, and return that at the end.

Reply:
It's my bad, I didn't catch this regression either.
I committed something slightly modified.